### ISSUE TITLE
[etherscan-verify] Support custom explorer URL

### ISSUE THREAD

**Issue by art049** [NomicFoundation/hardhat#2173]
Since https://github.com/blockscout/blockscout/pull/4908, Blockscout now comply to the Etherscan contract verification API.
It's now fairly easy to deploy a private testnets with an explorer that provides the same API as etherscan for the verification.

Thus, I think it might be interesting to be able configure the explorer url used for the verification on a custom network. 

An example implementation I have in mind would be to add an optional `explorer` field on each networks:
```typescript
const config: HardhatUserConfig = {
  networks: {
    privateNetwork: {
      chainId: 1337,
      explorer: {
        apiURL: "https://myprivate.network/api",
        browserURL: "https://myprivate.network"
      }
    },
};
```
At the moment, the only workaround I found was to modify `./packages/hardhat-etherscan/src/network/prober.ts` to manually add my custom explorer.

If you think it might be interesting, I should probably be able to implement this quickly .

**Comment by alcuadrado** [comment]
Thanks for this suggestion, @art049! It is a good idea.

We won't prioritize this right now, but if someone is willing to send a PR we can discuss it further.

**Comment by fvictorio** [comment]
I think the best way to do this is by moving the relevant logic to a subtask that can be overridden. It's not the best approach from the user's perspective, but it means we don't need to design how the config looks like, which is the hardest part of this feature.

**Comment by no2chem** [comment]
Hi @alcuadrado  - I've added an alternative PR in #2389 that achieves this by simply extending the chainConfig.ts

@fvictorio - I think simply adding an option to extend the chainConfig at runtime is sufficient - add a subtask is difficult because chainConfig is referenced from several locations directly.

**Comment by alcuadrado** [comment]
Wow! Thanks @no2chem! We'll review it asap

**Comment by fvictorio** [comment]
Closed by #2661.
